src/main/java/Duke.java

```diff
@@ -1,10 +1,294 @@
+        number = number.trim();
+        if (number.isBlank()) {
+            throw new DukeException("Please state which task you want to unmark.");
+        }
+        int index = Integer.parseInt(number) - 1;
+        if (index >= tasks.size() || index < 0) {
+            throw new DukeException("Please look carefully. This task is not inside the task list.");
+        }
+        tasks.get(index).unmark(horizontalLine);
+    }
+
+    public static Task createTask(String secondParts, String taskType) throws DukeException {
+        Task newTask = null;
+        if (secondParts.isBlank()) {
+            throw new DukeException("OOPS! The description of a " + taskType + " cannot be empty.");
+        }
+        secondParts = secondParts.trim();
+        switch (taskType) {
+        case "todo":
+            newTask = new ToDo(secondParts);
+            break;
+        case "deadline":
+            if (!secondParts.contains("by")) {
+                throw new DukeException("OOPS! 'by' keyword is missing. You are required "
+                        + "to state the deadline using the 'by' keyword.");
+            }
+            String[] instruction = secondParts.split(" by ", 2);
+            if (instruction.length < 2) {
+                throw new DukeException("OOPS! You forget to write do the task by when");
+            }
+            String description = instruction[0];
+            String deadline = instruction[1];
+            newTask = new Deadline(description, deadline);
+            break;
+        case "event":
+            if (!secondParts.contains("from") || !secondParts.contains("to")) {
+                throw new DukeException("OOPS! 'from' and/or 'to' keywords are missing. You are required to "
+                        + "state the starting and ending time using these two keywords.");
+            }
+            String[] e_instruction = secondParts.split(" from ", 2);
+            if (e_instruction.length < 2) {
+                throw new DukeException("OOPS! You forget to write the starting time of this event.");
+            }
+            String e_description = e_instruction[0];
+            String[] subInstruction = e_instruction[1].split(" to ", 2);
+            if (subInstruction.length < 2) {
+                throw new DukeException("OOPS! You forget to write the ending time of this event.");
+            }
+            String from = subInstruction[0];
+            String to = subInstruction[1];
+            newTask = new Event(e_description, from, to);
+            break;
+
+        }
+        return newTask;
+    }
+    public static void delete(String number, List<Task> tasks, String horizontalLine) throws DukeException {
+        number = number.trim();
+        if (number.isBlank()) {
+            throw new DukeException("Please state which task you want to delete.");
+        }
+        int index = Integer.parseInt(number) - 1;
```

Comment:
A bit of nitpicking but would `deleteIndex` be a better variable name in case you are working with other index variables in the same function in the future?